feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt

```diff
@@ -71,18 +72,16 @@ internal fun String.parsePriceUsingRegex(): Float? {
     val matcher = pattern.matcher(this)
     return if (matcher.find()) {
         val dirtyPrice = matcher.group()
-        var price: String
-        // 2 355 825.837
-        price = dirtyPrice.replace(" ", "")
-        val hasCommas = price.contains(",")
-        if (hasCommas) {
-            val numberOfCommas = price.length - price.replace(",", "").length
-            price = if (price.contains(".") || numberOfCommas > 1) {
-                // 1,000,000.00
-                price.replace(",", "")
+        var price = dirtyPrice.replace(" ", "")
+        val split = price.split(".", ",")
+        if (split.size != 1) {
+            // Assuming all prices we get have 2 decimal points
+            // Most currencies but Dirhan use 2 decimals. Amazon doesn't support Dirhan at the moment
+            price = if (split.last().length == 3) {
+                price.replace(".", "").replace(",", "")
             } else {
-                // 1,00
-                price.replace(",", ".")
+                val intPart = split.dropLast(1).joinToString("")
+                "$intPart.${split.last()}"
             }
         }
```

Comment:
I like this implementation much more. I also think it's easier to understand

Reply:
agreed